statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,508 @@
+        """
+        Returns calculation of mean.
+
+        Parameters
+        ----------
+        weights : np.array
+            The weights used to calculate the mean, will either be
+            original design weights or recalculated weights via jk,
+            boot, etc
+
+        Returns
+        -------
+        An array containing the statistic calculated on the columns
+        of the dataset.
+        """
+
+        # weights /= weights.sum()
+
+        return np.dot(weights, self.data) / np.sum(weights)
+
+
+class SurveyTotal(SurveyStat):
+    """
+    Calculates the total for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    method: string
+        User inputs whether to get bootstrap or jackknife SE
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the mean on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vc : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, se_method, replicates=None):
+        super().__init__(design)
+        self.design = design
+        self.data = np.asarray(data)
+
+        if se_method == "jack":
+            self.est, self.vc = super().jack(self)
+            self.vc = np.sqrt(np.diag(self.vc))
+        elif se_method == "boot":
+            self.est, self.vc = super().bootstrap(self, replicates)
+            self.vc = np.sqrt(self.vc)
+        else:
+            raise ValueError("Method %s not supported" % se_method)
+
+    def _stat(self, weights):
+        """
+        Returns calculation of mean.
+
+        Parameters
+        ----------
+        weights : np.array
+            The weights used to calculate the mean, will either be
+            original design weights or recalculated weights via jk,
+            boot, etc
+
+        Returns
+        -------
+        An array containing the statistic calculated on the columns
+        of the dataset.
+        """
+        return np.dot(weights, self.data)
+
+
+class SurveyQuantile(SurveyStat):
+    """
+    Calculates the qualtile[s] for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    parameter: array-like
+        array of quantiles to calculate for each column
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the quantiles on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    quantile : ndarray
+        The quantile[s] to calculate for each column
+    cumsum_weights : ndarray
```

Comment:
Attribute like this should be privatized (_cumsum_weights) and not documented.  Most users will not need this, and in future refactoring we don't want to be locked into providing this.